game/src/main/java/game/object/tank/TankWeapon.java

```diff
@@ -0,0 +1,38 @@
+package game.object.tank;
+
+import engine.geometry.Rectangle;
+import engine.geometry.Vector;
+
+public class TankWeapon {
+
+    private final static double RELATIVE_BULLET_SIZE = 0.2;
+    private final static double BULLET_MAX_SPEED = 5;
+
+    private final Rectangle tankBody;
+    private final Vector tankSpeed;
+
+    TankWeapon(Rectangle tankBody, Vector tankSpeed) {
+        this.tankBody = tankBody;
+        this.tankSpeed = tankSpeed;
+    }
+
+    public Rectangle getNewBulletBody() {
```

Comment:
what about?
```
public Bullet nextBullet() {

}
```

TankWeapon will also encapsulate bullet count. Each time you call nextBullet() bullet count will decrease for example.